### PR TITLE
Update/34063 store location skip btn

### LINKED ISSUE #34063: Store Location: Skip Button
## User Story

> As an experienced builder setting up a new store,
> I want to skip the onboarding wizard,
> So I can manually set up the store later.

## Acceptance Criteria

```
Feature: Onboarding Profiler

  Scenario: User views the onboarding wizard
    When the user views any page of the onboarding wizard
    Then a small button labelled "Skip" displays below the form
    And a tooltip button appears next to the skip button

  Scenario: User skips the onboarding wizard
    When a user skips the onboarding wizard
    Then the user is asked to confirm usage tracking
    Then the user is redirected to WooCommerce Home Screen.
 
  Scenario: User has unsaved data
    Given unsaved data exists on the form
    When a user skips the onboarding wizard
    Then the user is prompted with a confirm dialog 
    And the dialog text is "You're about to skip store setup with unsaved changes. Do you want to continue"
    And the dialog buttons are "Cancel" and "Skip setup"
   
   Scenario: User clicks on the tooltip button
    When a user clicks on the tooltip button
    Then a tooltip displays with the text "Manual setup is only recommended for experienced WooCommerce users or developers."
```

## Screenshots

image

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #34063

This PR adds `Skip` button on the OBW steps. 

### How to test the changes in this Pull Request:

Make sure `woocommerce_allow_tracking` option is set to `no`

1. Start with a fresh site.
2. Start OBW
3. You should see `Skip` button at the bottom of each step. 
4. Click exclamation mark. A popover should appear with text `manual setup is only recommended for experienced WooCommerce users or developers.`
5. Click `Skip` button. UsageModal should show up.
6. Click "No Thanks`
7. Confirm `woocommerce_allow_tracking` is still `no`
8. Go back to the OBW
9. Click `Skip` button again.
10. Click `Yes, count me in` button.
11. You should be redirected to `WooCommerce -> Home`
12. Confirm `woocommerce_allow_tracking` is now `yes`

**Testing tracks**

Set `woocommerce_allow_tracking` to `no`

1. Go back to OBW again
2. Open browser inspector and enable tracks debug.
3. Click `Skip` button on each page and click `Yes, count me in`
4. Confirm the track event is recorded.

### Screenshots

**UsageModal**:
Screen Shot 2022-08-04 at 11 54 25 AM

### Other information:

-   [X] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [X] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by chihsuan** [comment]
> 3. Click Skip button on each page and click Yes, count me in
> 4. Confirm the track event is recorded.

@moon0326 I observed that the track event is recorded four times on my end. 🤔  Does that same happen to you?

https://user-images.githubusercontent.com/4344253/182996389-3635c7e2-5965-4db7-a214-74fab085c58d.mov

**Comment by moon0326** [comment]
> > 3. Click Skip button on each page and click Yes, count me in
> > 4. Confirm the track event is recorded.
> 
> @moon0326 I observed that the track event is recorded four times on my end. 🤔 Does that same happen to you?
> 
>  Screen.Recording.2022-08-05.at.11.36.18.mov

Thank you for catching it. Yes, I can also see that on my end. I'll look into it 👍

**Comment by moon0326** [comment]
@chihsuan Fixed in https://github.com/woocommerce/woocommerce/pull/34176/commits/42f6c6a23874dd7b1e94f53b7f7212590f4cac0f

A little explanation..

So [UsageModal](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/profile-wizard/steps/usage-modal.js#L102) calls `onClose` and `onContinue` from `SkipButton` before finishing rendering. The problem is that `onClose` changes a [state](https://github.com/woocommerce/woocommerce/blob/update/34063-store-location-skip-btn/plugins/woocommerce-admin/client/profile-wizard/steps/skip-button.tsx#L74) of `SkipButton` . This causes `cannot update a component while rendering a different cmponent` warning.

I used `setTimeout` trick to get around it, but it caused multiple executions of the same callback.

I fixed it by calling `onClose` and `onContinue` directly from this [callback](https://github.com/woocommerce/woocommerce/pull/34176/commits/414f543a9a25876ab71670523e44d815057138b6) instead.

File: plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js [hunk 1]
```diff
@@ -112,10 +110,13 @@ export class StoreDetails extends Component {
 	}
 
 	onSubmit() {
-		this.setState( {
-			showUsageModal: true,
-			skipping: false,
-		} );
+		if ( this.props.allowTracking ) {
+			this.props.skipProfiler();
```
Comment (chihsuan) [inline plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js]: @moon0326 There is an edge case. When I clicked on "Continue" button and  `allowTracking = true`, I was redirected to the home page. Is this expected?

Steps to reproduce:

1. Go to OBW
2. Skip the profiler and allow tracking
3. Go to OBW again
4. Select a country
5. Click "Continue" button


https://user-images.githubusercontent.com/4344253/183330456-d015aac0-4c46-433a-8c24-e011558c2099.mov

**Review by chihsuan — CHANGES_REQUESTED** [review]

File: plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce-admin/client/profile-wizard/steps/store-details/index.js]: Thank you for catching it 👍 

The method should have been `goToNextStep` instead. Updated in https://github.com/woocommerce/woocommerce/pull/34176/commits/8089eac4c0f41fd48f8151222d41b2d3fbc986b3

**Review by chihsuan — APPROVED** [review]
Nice work! Tested well and LGTM. 👍

**Pull request merged**